R/openapi-spec.R

```diff
@@ -61,17 +61,7 @@ responsesSpecification <- function(endpts){
   }
   for (resp in names(resps)) {
     if (!length(resps[[resp]]$content)) {
-      ctype <- NULL
-      if (is.function(endpts$serializer)) {
-        ctype <- formals(endpts$serializer)$type
-        if (is.null(ctype)) {
-          ctype <- tryCatch({
-            get("type", envir =
-                  environment(get("serialize_fn", envir =
-                                    environment(endpts$serializer))))},
-            error = function(e) {NULL})
-        }
-      }
+      ctype <- environment(endpts$serializer)$headers$`Content-Type`
```

Comment:
```suggestion
      if (is.function(endpts$serializer)) {
        # Must safe-guard against partial name matching 
        # since we are reaching into the function env
        ctype <- environment(endpts$serializer)[["headers"]][["Content-Type"]]
      }
```